opentelemetry-sdk/tests/performance/benchmarks/trace/propagation/test_benchmark_b3_format.py

```diff
@@ -0,0 +1,42 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+
+import opentelemetry.sdk.trace as trace
+import opentelemetry.sdk.trace.propagation.b3_format as b3_format
+from opentelemetry.trace.propagation.textmap import DictGetter
+
+FORMAT = b3_format.B3Format()
+
+
+@mock.patch(
+    "opentelemetry.sdk.trace.propagation.b3_format.trace.get_tracer_provider",
+    new=trace.TracerProvider,
+)
+def test_extract_single_header(benchmark):
+    benchmark(
+        FORMAT.extract,
+        DictGetter(),
+        {
+            FORMAT.SINGLE_HEADER_KEY: "bdb5b63237ed38aea578af665aa5aa60-00000000000000000c32d953d73ad225"
+        },
+    )
+
+
+def test_inject_empty_context(benchmark):
+    tracer = trace.TracerProvider().get_tracer("sdk_tracer_provider")
+    with tracer.start_as_current_span("Root Span"):
+        with tracer.start_as_current_span("Child Span"):
+            benchmark(FORMAT.inject, dict.__setitem__, {})
```

Comment:
Should we be benchmarking a full context instead?

Reply:
Sure I can add one, but I don't think there are any specs on this.

I noticed that on Java they don't have a complicated context as seen [here](https://github.com/open-telemetry/opentelemetry-java/blob/v0.13.x/extensions/trace-propagators/src/jmh/java/io/opentelemetry/extension/trace/propagation/PropagatorContextInjectBenchmark.java#L54).

But if we want I can add a context like this one I saw in the [b3 unit tests](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/tests/trace/propagation/test_b3_format.py#L80-L85)

Reply:
I don't think there are any specs defined for benchmark testing in general. The purpose is to test the performance of the worst case performance impact. In this case it would be a full context with all fields defined properly and injected. As long as this condition is met it doesn't matter where you get it from.

But to answer your question, the b3 unit test span contexts looks appropriate.

Reply:
Sounds good! I've added a context to the `inject` tests.